moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -541,15 +541,16 @@ double JogCalcs::decelerateForSingularity(const Eigen::VectorXd& commanded_veloc
   return velocity_scale;
 }
 
-void JogCalcs::enforceJointVelocityLimits(Eigen::VectorXd& calculated_joint_change)
+void JogCalcs::enforceJointVelocityLimits(Eigen::ArrayXd& calculated_joint_change)
 {
   double maximum_joint_change = calculated_joint_change.cwiseAbs().maxCoeff();
-  if (maximum_joint_change > parameters_.joint_scale * parameters_.publish_period)
+  if ((maximum_joint_change / parameters_.publish_period) > parameters_.max_joint_velocity)
   {
     // Scale the entire joint velocity vector so that each joint velocity is below min, and the output movement is
-    // scaled uniformly to match expected motion
-    calculated_joint_change =
-        calculated_joint_change * parameters_.joint_scale * parameters_.publish_period / maximum_joint_change;
+    // scaled uniformly to match direction of expected motion.
+    // The formula is delta_j(i) = delta_j(i) / (|max_velocity| / velocity_limit)
+    calculated_joint_change = calculated_joint_change /
+                              ((maximum_joint_change / parameters_.publish_period) / parameters_.max_joint_velocity);
```

Comment:
Are there checks that all these variables are non-zero? Asserts at top of function would be nice

Reply:
Added a check for publish_period > 0 in jog_interface_base.cpp.

max_joint_velocity already got checked.